lib/wasi/src/state/pipe.rs

```diff
@@ -93,26 +234,108 @@ impl WasiPipe {
     }
 }
 
+impl Write for WasiPipe {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
```

Comment:
At some point we should probably add backpressure so that very large writes do not use up all the memory - for now lets not worry about it unless you are feeling up to the challenge.

Reply:
Tracking in https://github.com/wasmerio/wasmer/issues/3191, this PR is already pretty old and I just want to get this merged.